### ISSUE TITLE
Add Hardware-inventory collector for Hardware Status

### ISSUE THREAD

**Issue by SckyzO** [treydock/eseries_exporter#13]
Hello @treydock 

Thx for your work. I use your exporter and it work perfectly. 

Do you think it is possible to add hardware support in your collector to check powersupplies / fans ... etc ? 
It would be available at https://localhost:8443/devmgr/v2/storage-systems/<ID>/hardware-inventory

hhmmm, another thing, in your readme, you use a "simple id" in your example, but when you are in production and use autodiscover function, Eseries ID will be automatically generated (like this : https://localhost:8443/devmgr/v2/storage-systems/b81debff-4fr5-43a3-a79c-57246a55f5a3)

I use prometheus configuration like this : 
```
  - job_name: eseries_da30
    metrics_path: /eseries
    static_configs:
    - targets: ['b81debff-4fr5-43a3-a79c-57246a55f5a3']
    relabel_configs:
    - source_labels: [__address__]
      target_label: __param_target
    - source_labels: [__param_target]
      target_label: instance
      replacement: eserie_01
    - target_label: __address__
      replacement: 127.0.0.1:9313
```

So i think it will be a good way to add "name" support

example : 
https://localhost:8443/devmgr/v2/storage-systems
```json
[
  {
    "id": "b81debff-4fr5-43a3-a79c-57246a55f5a3",
    "name": "eserie_01",
    "wwn": "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx",
    "passwordStatus": "valid",
    "passwordSet": true,
    "status": "optimal",
    "certificateStatus": "trusted",
```

Thank a lot, have a nice day

--
Tom

**Comment by treydock** [comment]
@SckyzO I will look at adding more hardware inventory data.

As for the ID used, I wonder if maybe the difference with how you supply the ID is based on how the web API proxy was setup?  This is an example of how I can query my endpoint:

```
curl -u <OMIT> http://localhost:8080/devmgr/v2/storage-systems/e5660-01/hardware-inventory | jq .controllers
```
The ID I get is literally the name of the controller:

```
[root@prometheus01 ~]# curl -q -u <OMIT> http://localhost:8080/devmgr/v2/storage-systems 2>/dev/null | jq .[0].id
"e5660-01"
```

Looking at my notes I think the ID I get is based on how I setup each controller with the web API proxy:

```
curl -X POST -u admin:OMIT "http://localhost:8080/devmgr/v2/storage-systems" -H  "accept: application/json" -H  "Content-Type: application/json" -d "{  \"id\": \"e5660-01\",  \"controllerAddresses\": [    \"10.10.2.101\" , \"10.10.2.102\" ],  \"acceptCertificate\": true,  \"validate\": false,  \"password\": \"OMIT\"}"
```

Sort of called out in README but not a great example: https://github.com/treydock/eseries_exporter#dependencies

**Comment by treydock** [comment]
@SckyzO Try https://github.com/treydock/eseries_exporter/releases/tag/v1.1.0-rc.0 - this has hardware inventory added

**Comment by SckyzO** [comment]
Wow ! What reactivity !!! 😍
I'm testing this and getting back to you tomorrow.

Thanks again

Tom

**Comment by SckyzO** [comment]
Hardware Inventory Work perfectly. Thank you @treydock 👍

> As for the ID used, I wonder if maybe the difference with how you supply the ID is based on how the web API proxy was setup? 
Yes it is. We use autodiscover, so IDs are autogenerated by web API. I think i'm not alone to use this function :) 

It would be good, perhaps, to add in the README how to retrieve the IDs and if necessary "rename" them in prometheus.
Here is my prometheus conf and the command line to retrieve the IDs sorted by "name"

```bash
curl -X GET -u <username>:<password> "http://localhost:8081/devmgr/v2/storage-systems" -H "accept: application/json"  2>/dev/null | jq -r '.[] | "ID: \(.id)\tname: \(.name)\tip1: \(.ip1)\tip2: \(.ip2)"'

ID: d7agcx01-6950-4dc9-b802-4mpq2e03p4kq        name: eserie_01      ip1: 192.168.1.1    ip2: 192.168.1.2
ID: h81mpsdf-4ffd-43a3-a79c-57246a55f5e6        name: eserie_02      ip1: 192.168.1.3    ip2: 192.168.1.4
ID: 9e1cdf4c-6jed-4c39-8f20-9896gh3969f7        name: eserie_03      ip1: 192.168.1.5    ip2: 192.168.1.6
ID: eb1e6e3p-7643-44c4-9604-9e9783dd6124        name: eserie_04      ip1: 192.168.1.7    ip2: 192.168.1.8
```

Add configure prometheus like this : 
``` 
  - job_name: eserie_01
    metrics_path: /eseries
    static_configs:
    - targets: ['d7agcx01-6950-4dc9-b802-4mpq2e03p4kq']
    relabel_configs:
    - source_labels: [__address__]
      target_label: __param_target
    - source_labels: [__param_target]
      target_label: instance
      replacement: eserie_01
    - target_label: __address__
      replacement: 127.0.0.1:9313

  - job_name: eserie_02
    metrics_path: /eseries
    static_configs:
    - targets: ['h81mpsdf-4ffd-43a3-a79c-57246a55f5e6']
    relabel_configs:
    - source_labels: [__address__]
      target_label: __param_target
    - source_labels: [__param_target]
      target_label: instance
      replacement: eserie_02
    - target_label: __address__
      replacement: 127.0.0.1:9313

[...]

```

**Comment by treydock** [comment]
Just FYI that one way to simplify your configs since usually better to have one job per type of scrape vs one job per target.  This allows you to specify multiple targest, add a name label, and then assign that name label to instances and then remove the extra name label.


```
- job_name: eseries
    metrics_path: /eseries
    static_configs:
    - targets:
       - 'd7agcx01-6950-4dc9-b802-4mpq2e03p4kq'
      labels:
        name: eserie_01
    - targets:
       - 'h81mpsdf-4ffd-43a3-a79c-57246a55f5e6'
      labels:
        name: eserie_02
    relabel_configs:
    - source_labels: [__address__]
      target_label: __param_target
    - target_label: __address__
      replacement: 127.0.0.1:9313
    - source_labels: [name]
      target_label: instance
    - regex: '^(name)$'
      action: labeldrop
```

Added example on querying names here: https://github.com/treydock/eseries_exporter#dependencies

Added another example like the above here: https://github.com/treydock/eseries_exporter#prometheus-configs

**Comment by SckyzO** [comment]
Ho great 😁 thanks

**Comment by SckyzO** [comment]
Just for say, if you add : 
```
    - regex: '^(name)$'
      action: labeldrop
```
prometheus try to resolv localhost:9313:/eseries?target=eserie_01 (but doesn't exist), 
So juste I removed this ;-)

And, for keep "instance=eserie_id", I use this  (with your conf above, i had "instance=localhsot:9313"): 

```
    relabel_configs:
    - source_labels: [__address__]
      target_label: __param_target
    - target_label: __address__
      replacement: 127.0.0.1:9313
    - source_labels: [__param_target]
      target_label: instance
    - source_labels: [name]
      target_label: name
```

Thank you

**Comment by treydock** [comment]
Can you provide full job config? Because what you described should be not necessary. I use the `labeldrop` pattern to set parameters for scrape jobs and then drop the label and never ran into issues.  Also the source and target being same label does nothing, the purpose of relabel is to change labels so if both source and target are the same label, it does nothing.

**Comment by SckyzO** [comment]
Last information
Your exporter work fine (tested) : SANtricity Web Proxy with versions 4 and versions 5 :-)

**Comment by SckyzO** [comment]
I think issue can be closed and RC release can be the new Final release :)

**Comment by treydock** [comment]
Released 1.1.0 with this change.
